js/private/postinstall.js

```diff
@@ -0,0 +1,57 @@
+const {mkdirSync, copyFileSync, readdirSync, readFileSync, existsSync, statSync} = require('fs');
```

Comment:
Did you look at existing implementation of lifecycle in npm cli?

Reply:
I didn't think to check. I'll do that. I realize the implementation will need to become a lot more complex, but just wanted to do a simple pass first.

Reply:
So the npm cli ends up calling a spawn like we do here, but they also add node_modules' `.bin` dirs to the path, and they have some windows-specific logic around escaping arguments and how the path gets set, whether `cmd.exe` gets called, etc.

My plan was to work on adding `.bin`s to the PATH next. Are you okay with this simpler version going in first?

Reply:
yes, just wanted you to get some context from their impl to inform TODOs, comments we could add here about design, etc.

I'd at least suggest adding some pointers to what you found, so that when we read this code later we can compare with npm